src/PokemonContainer.js

```diff
@@ -0,0 +1,123 @@
+import { Component } from 'react';
+import PokemonList from './PokemonList';
+import Loader from "react-loader-spinner";
+
+
+import './App.css';
+
+class PokemonContainer extends Component {
+  state = { data: [], loading: true, query: null, sortOrder: 'asc', page: 1 };
+
+  fetchPokemon = async () => {
+    let url = 'https://pokedex-alchemy.herokuapp.com/api/pokedex/';
+    
+    let searchParams = new URLSearchParams();
+    searchParams.set('page', this.state.page);
+    searchParams.set('perPage', 20);
+    
+    if (this.state.query) {
+     searchParams.set('pokemon', this.state.query);
+    }
+    
+    if (this.state.sortOrder) {
+      searchParams.set('sort', 'species_id');
+      searchParams.set('direction', this.state.sortOrder);
+    }
+    
+    url = url + `?${searchParams.toString()}`;
+
+    
+    let response = await fetch(url);
+    
+    let data = await response.json();
+    
+    setTimeout(() => {
+      this.setState({ data, loading: false });
+
+    }, 3000)
```

Comment:
You don't really need this timeout here since the API call is already asynchronous. 

Also, I would recommend saving data.results directly to your state rather than the full data object you receive back from the API call since that contains extra data that you don't actually need for your app to function. You want to minimize the amount of data you're storing in your app as much as possible to help with performance.